packages/node/src/app/analytics-node.ts

```
@@ -0,0 +1,345 @@
+   * @param properties
+   * @param options
+   * @param callback
+   */
+  screen(
+    name: string,
+    properties: object,
+    options: SegmentEventOptionsNode,
+    callback?: Callback
+  ): Promise<ContextNode>
+
+  screen(...args: PageParams): Promise<ContextNode> {
+    const [category, page, properties, options, callback] =
+      resolvePageArguments(...args)
+
+    const segmentEvent = this.eventFactory.screen(
+      category,
+      page,
+      properties,
+      options,
+      this.integrations
+    )
+
+    return dispatch(segmentEvent, this.queue, this, { callback })
+      .then((ctx) => {
+        this.emit(
+          'page',
+          category,
+          page,
+          ctx.event.properties,
+          ctx.event.options
+        )
+        return ctx
+      })
+      .catch((ctx) => ctx)
+  }
+  /**
+   * Records actions your users perform.
+   * @param event - The name of the event you're tracking.
+   * @param properties - A dictionary of properties for the event.
+   * @param options
+   * @param callback
+   */
+  track(
+    event: string,
+    properties: object,
+    options: SegmentEventOptionsNode,
+    callback?: Callback
+  ): Promise<ContextNode> {
+    const segmentEvent = this.eventFactory.track(
+      event,
+      properties as EventProperties,
+      options,
+      this.integrations
+    )
+
+    return dispatch(segmentEvent, this.queue, this, {
+      callback,
+    })
+      .then((ctx) => {
+        this.emit('track', event, ctx.event.properties, ctx.event.options)
+        return ctx
+      })
+      .catch((ctx) => ctx)
+  }
+
+  /**
+   * Registers one or more plugins to augment Analytics functionality.
+   * @param plugins
+   */
+  async register(...plugins: CorePlugin<any, any>[]): Promise<ContextNode> {
+    const ctx = ContextNode.system()
+
+    const registrations = plugins.map((xt) =>
+      this.queue.register(ctx, xt, this)
+    )
+    await Promise.all(registrations)
```

we might want to swallow these errors, but left as is